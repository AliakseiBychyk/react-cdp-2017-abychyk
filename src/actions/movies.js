export const RECEIVE_MOVIES = 'RECEIVE_MOVIES'

export const receiveMovies = (movies) => ({
  type: RECEIVE_MOVIES,
  movies  
})

export const fetchMoviesJson = (criterion, query) => { 
  const url = `https://api.themoviedb.org/3/search/${criterion}?api_key=${api_key}&query=${query}&include_adult=true`   

  return fetch(url, {
    method: 'GET'
  }).then(resp => resp.json())
    .then(data => {
      return criterion === 'movie'
        ? data.results
        : data.results[0].known_for
    })
    .catch(console.error)
}