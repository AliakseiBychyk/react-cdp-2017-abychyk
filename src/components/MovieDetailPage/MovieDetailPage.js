import React, { Component } from 'react'
import MovieItem from '../MovieItem/MovieItem'
import MovieList from '../MovieList/MovieList'
import styles from './MovieDetailPage.css'

class MovieDetailPage extends Component {
  componentWillMount() {
    if (this.props.movies.length === 0) {
      this.props.fetchMovies('title', encodeURIComponent(this.props.title))
    }   
  }

  render() {
    return (
      <div className={styles.detail}>
        {this.props.movie ? <MovieItem
          movie={this.props.movie}
        /> : null}
        <MovieList
          movies={this.props.movies}
        />
      </div>
    )
  }
}

export default MovieDetailPage