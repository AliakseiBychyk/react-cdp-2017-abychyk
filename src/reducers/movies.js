import { RECEIVE_MOVIES } from '../actions/movies'

export default (state = [], action) => {
  switch (action.type) {
    
    case RECEIVE_MOVIES:
      return action.movies

    default:
      return state
  }
}