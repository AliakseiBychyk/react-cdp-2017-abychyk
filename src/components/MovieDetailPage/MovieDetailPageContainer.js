import { connect } from 'react-redux'
import { fetchMovies } from '../../actions/movies'
import MovieDetailPage from './MovieDetailPage'

const mapStateToProps = (state, props) => {
  return {
    movies: state.movies,
    movie: state.movies.find(movie => 
     movie.title == props.match.params.title
    )
  }
}

export default connect(mapStateToProps, { fetchMovies })(MovieDetailPage)