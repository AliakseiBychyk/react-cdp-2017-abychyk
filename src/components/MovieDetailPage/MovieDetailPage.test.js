import React from 'react'
import { shallow } from 'enzyme'
import MovieDetailPage from './MovieDetailPage'
import movies from '../../fixtures/testingMovieQuery.json'

let fetchMovies

beforeEach(() => {
  fetchMovies = jest.fn()
})

test('should render MovieDetailPage correctly', () => {
  const wrapper = shallow(<MovieDetailPage
    movies={[]}
    fetchMovies={fetchMovies} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render MovieDetailPage with movies correctly', () => {
  const wrapper = shallow(<MovieDetailPage
    movie={movies.results[1]}
    movies={movies.results}
    fetchMovies={fetchMovies} />)
  expect(wrapper).toMatchSnapshot()
})