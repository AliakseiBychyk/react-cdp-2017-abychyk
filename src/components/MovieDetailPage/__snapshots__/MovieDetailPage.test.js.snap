// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieDetailPage correctly 1`] = `
<div
  className="detail"
>
  <MovieItem />
  <MovieList
    movies={Array []}
  />
</div>
`;

exports[`should render MovieDetailPage with movies correctly 1`] = `
<div
  className="detail"
>
  <MovieItem
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/mSE7vKEiWY1NFbvrJIwrcHOtiSH.jpg",
        "genre_ids": Array [
          80,
          9648,
          53,
        ],
        "id": 3093,
        "original_language": "en",
        "original_title": "Basic Instinct 2",
        "overview": "Novelist Catherine Tramell is once again in trouble with the law, and Scotland Yard appoints psychiatrist Dr. Michael Glass to evaluate her. Though, like Detective Nick Curran before him, Glass is entranced by Tramell and lured into a seductive game.",
        "popularity": 15.958979,
        "poster_path": "/gXzrKzHnMmfWCF5PMYQfNOzCxYl.jpg",
        "release_date": "2006-03-29",
        "title": "Basic Instinct 2",
        "video": false,
        "vote_average": 4.6,
        "vote_count": 183,
      }
    }
  />
  <MovieList
    movies={
      Array [
        Object {
          "adult": false,
          "backdrop_path": "/bKdxJMqvr4wltzAQaieA6CTkaa9.jpg",
          "genre_ids": Array [
            53,
            9648,
          ],
          "id": 402,
          "original_language": "en",
          "original_title": "Basic Instinct",
          "overview": "A police detective is in charge of the investigation of a brutal murder, in which a beautiful and seductive woman could be involved.",
          "popularity": 11.542996,
          "poster_path": "/p1DHOfLFZmM567ePKfFNMV4mY92.jpg",
          "release_date": "1992-03-20",
          "title": "Basic Instinct",
          "video": false,
          "vote_average": 6.5,
          "vote_count": 859,
        },
        Object {
          "adult": false,
          "backdrop_path": "/mSE7vKEiWY1NFbvrJIwrcHOtiSH.jpg",
          "genre_ids": Array [
            80,
            9648,
            53,
          ],
          "id": 3093,
          "original_language": "en",
          "original_title": "Basic Instinct 2",
          "overview": "Novelist Catherine Tramell is once again in trouble with the law, and Scotland Yard appoints psychiatrist Dr. Michael Glass to evaluate her. Though, like Detective Nick Curran before him, Glass is entranced by Tramell and lured into a seductive game.",
          "popularity": 15.958979,
          "poster_path": "/gXzrKzHnMmfWCF5PMYQfNOzCxYl.jpg",
          "release_date": "2006-03-29",
          "title": "Basic Instinct 2",
          "video": false,
          "vote_average": 4.6,
          "vote_count": 183,
        },
        Object {
          "adult": false,
          "backdrop_path": null,
          "genre_ids": Array [
            99,
          ],
          "id": 258826,
          "original_language": "en",
          "original_title": "Between the Sheets: A Look Inside 'Basic Instinct 2'",
          "overview": "A Look Inside 'Basic Instinct 2'",
          "popularity": 1.007875,
          "poster_path": null,
          "release_date": "",
          "title": "Between the Sheets: A Look Inside 'Basic Instinct 2'",
          "video": false,
          "vote_average": 0,
          "vote_count": 0,
        },
        Object {
          "adult": false,
          "backdrop_path": null,
          "genre_ids": Array [
            99,
          ],
          "id": 258827,
          "original_language": "en",
          "original_title": "Blonde Poison: The Making of 'Basic Instinct'",
          "overview": "A look behind the scenes at one of the more controversial thillers of the 90's, and the one that made Sharon Stone a top-name star. Includes interviews with Stone and footage of some of the hard work that went into making the movie, as well as some discussion of the problems the makers had to deal with, including protests from groups who wanted the film changed or simply abandoned.",
          "popularity": 1.007685,
          "poster_path": null,
          "release_date": "",
          "title": "Blonde Poison: The Making of 'Basic Instinct'",
          "video": false,
          "vote_average": 8,
          "vote_count": 1,
        },
      ]
    }
  />
</div>
`;
